Source/Core/Core/HW/DSPHLE/UCodes/AXWiiOld.cpp

```diff
@@ -0,0 +1,190 @@
+// Copyright 2008 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+//
+#define AX_WII  // Used in AXVoice.
+#define AX_WII_OLD
+
+#include "Core/HW/DSPHLE/UCodes/AXWiiOld.h"
+
+#include <algorithm>
+#include <array>
+
+#include "Common/ChunkFile.h"
+#include "Common/CommonTypes.h"
+#include "Common/Logging/Log.h"
+#include "Common/Swap.h"
+#include "Core/HW/DSPHLE/DSPHLE.h"
+#include "Core/HW/DSPHLE/MailHandler.h"
+#include "Core/HW/DSPHLE/UCodes/AXStructs.h"
+#include "Core/HW/DSPHLE/UCodes/AXVoice.h"
+#include "Core/HW/DSPHLE/UCodes/UCodes.h"
+
+namespace DSP::HLE
+{
+AXWiiOldUCode::AXWiiOldUCode(DSPHLE* dsphle, u32 crc) : AXWiiUCode(dsphle, crc)
+{
+  INFO_LOG(DSPHLE, "Instantiating AXWiiOldUCode");
+}
+
+AXWiiOldUCode::~AXWiiOldUCode()
+{
+}
+
+void AXWiiOldUCode::HandleCommandList()
+{
+  // Temp variables for addresses computation
+  u16 addr_hi, addr_lo;
+  u16 addr2_hi, addr2_lo;
+  u16 volume;
+
+  u32 pb_addr = 0;
+
+  // WARN_LOG(DSPHLE, "Command list:");
+  // for (u32 i = 0; m_cmdlist[i] != CMD_END; ++i)
+  //     WARN_LOG(DSPHLE, "%04x", m_cmdlist[i]);
+  // WARN_LOG(DSPHLE, "-------------");
+
+  u32 curr_idx = 0;
+  bool end = false;
+  while (!end)
+  {
+    u16 cmd = m_cmdlist[curr_idx++];
+
+    switch (cmd)
```

Comment:
This whole switch statement looks quite duplicated, is there a way the common parts could be abstracted into a template?

Reply:
I don't think so. All the enum constants have an offset and there's some other subtle differences.

Reply:
(I know this discussion was ended on IRC yesterday, leaving it here just in case)

https://godbolt.org/z/p-3Dvf